Rakefile

```diff
@@ -114,24 +112,28 @@ task :benchmark do
   fake_url = 'http://www.google.com' #resolve but unable to send I hope!
 
   empty_app = EmptyMiddleware.new(nil)
-  app = FaradayMiddleware.new(nil)
 
   null_configuration = { logger: logger, sample_rate: 1}
   json_configuration = null_configuration.merge({ json_api_host: fake_url })
   scribe_configuration = null_configuration.merge({ scribe_server: fake_url })
 
+  # We create a different faraday middleware per rack middleware below because
+  # both middlewares share the same tracer. So they need to be created in pairs.
   empty_rack = EmptyMiddleware.new(empty_app)
```

Comment:
I'm doing two different things there depending on if we pass an app or nil, maybe should be refactored on two classes